client/lib/plans/index.js

```diff
@@ -69,8 +69,35 @@ export function getPlanPath( plan ) {
 	return get( getPlan( plan ), 'getPathSlug', () => undefined )();
 }
 
+/**
+ * Determines if a plan has a specific feature.
+ *
+ * Collects features for a plan by calling all possible feature methods for the plan.
+ *
+ * @param   {Object|String} plan    Plan object or plan name.
+ * @param   {String}        feature Feature name.
+ * @returns {Boolean}               Whether the specified plan has the specified feature.
+ */
 export function planHasFeature( plan, feature ) {
-	return includes( get( getPlan( plan ), 'getFeatures', () => [] )(), feature );
+	const planConstantObj = getPlan( plan );
+
+	// Collect features from all plan methods (may have duplicates)
+	const allFeatures = [
+		'getPlanListFeatures',
+		'getPromotedFeatures',
+		'getSignupFeatures',
+		'getBlogSignupFeatures',
+		'getPortfolioSignupFeatures',
+		'getHiddenFeatures',
+	].reduce(
+		( featuresArray, featureMethodName ) => [
+			...get( planConstantObj, featureMethodName, () => [] )(),
+			...featuresArray,
+		],
+		[]
+	);
+
+	return includes( allFeatures, feature );
```

Comment:
One risk in this refactor is having this function return a different result if testing for features that are not in the `getFeatures/getPlanListFeatures` function, but are in other features functions (like `getPromotedFeatures`, `getSignupFeatures`, etc), since all of those additional functions are now included in the feature checking.

I did thorough checking of the `feature` argument at all calls of `planHasFeature` (both calling this function directly and through `hasFeature` selector in `client/state/sites/plans/selectors.js`). All calls use features that are listed in `getFeatures/getPlanListFeatures`, and thus the return value will be the same before and after this refactor.